### PR TITLE
Habitat 2.0 Tasks Cleanup

### PR DESCRIPTION
**Pull request by ASzot**
## Motivation and Context
Various improvements to Habitat 2.0 tasks:
* Speed up simulator `reset` by directly loading the asset path rather than searching by matching substring. Thank you to @JHurricane96  for originally implementing this change. 
* Change `recurrent_generator` -> `data_generator` since not all rollout storage will generate "recurrent policy" data.
* Fixed bug with objects not being properly removed. Thank you @yvsriram for originally implementing this change.
* Added support for predicate to set objects on top of receptacles.
* Fixed a bug in the predicate action.
* Allowed setting the level of the Habitat logger via an environment variable like our other loggers.
* Removed `env` and `pth` metrics and added more detailed timer information to `ppo_trainer`.

## How Has This Been Tested

Training unaffected locally. 

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation
- **\[Refactoring\]** Large changes to the code that improve its functionality or performance
- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

### CODE REVIEW

File: habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py [hunk 1]
```diff
@@ -223,7 +225,10 @@ def _broadcast_skill_ids(
         for k, v in skill_to_batch.items():
             grouped_skills[k] = (
                 v,
-                {dat_k: dat[v] for dat_k, dat in sel_dat.items()},
+                {
+                    dat_k: dat if dat is None else dat[v]
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py:229]: Are we okay with having dat be Any ? In which cases can this be null ? 
```suggestion
                    dat_k: None if dat is None else dat[v]
```

File: habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py [hunk 2]
```diff
@@ -572,11 +572,9 @@ def _training_log(
             )
 
             logger.info(
-                "update: {}\tenv-time: {:.3f}s\tpth-time: {:.3f}s\t"
+                "update: {}\t"
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py:575]: use an f string :   
`f"Num updates: {self.num_updates_done}\tNum frames {self.num_steps_done}"`

**Review by vincentpierre — APPROVED** [review]
Thanks @ASzot If the tests pass I am ok with merging.

File: habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py (hunk 1 above)
Comment (ASzot) [reply habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py:229]: Actually it's okay to leave how it was before. Thanks for pointing this out.

**Pull request merged**